File: frontend/src/lib/api/recruitment.ts
import { request, requestUpload } from "./client";

export interface RecruitmentPosition {
    id: string;
    title: string;
    department: string | null;
    description: string | null;
    required_skills: string[];
    experience_min_years: number;
    salary_range_min: number | null;
    salary_range_max: number | null;
    status: string;
    candidate_count: number;
}

export interface Candidate {
    id: string;
    position_id: string | null;
    name: string;
    email: string | null;
    phone: string | null;
    skills: string[];
    experience_years: number | null;
    languages: { lang: string; level: string }[];
    education: string | null;
    summary: string | null;
    score: number | null;
    score_breakdown: Record<string, number> | null;
    status: string;
    cv_file_path: string | null;
}

export const recruitment = {
    listPositions: (status = "all") =>
        request<RecruitmentPosition[]>(`/api/v1/recruitment/positions?status_filter=${status}`),

    createPosition: (data: {
        title: string;
        department?: string;
        description?: string;
        required_skills?: string[];
        experience_min_years?: number;
        salary_range_min?: number;
        salary_range_max?: number;
    }) => request<RecruitmentPosition>("/api/v1/recruitment/positions", {
        method: "POST",
        body: JSON.stringify(data),
    }),

    listCandidates: (positionId: string) =>
        request<Candidate[]>(`/api/v1/recruitment/positions/${positionId}/candidates`),

    uploadCV: (positionId: string, file: File): Promise<Candidate> => {
        const form = new FormData();
        form.append("file", file);
        return requestUpload<Candidate>(`/api/v1/recruitment/positions/${positionId}/upload-cv`, form);
    },

    updateCandidateStatus: (candidateId: string, status: string) =>
        request<Candidate>(`/api/v1/recruitment/candidates/${candidateId}/status`, {
            method: "PATCH",
            body: JSON.stringify({ status }),
        }),
};

File: backend/app/api/v1/schemas/recruitment.py
"""Pydantic schemas for recruitment endpoints."""

from typing import Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field, model_validator


class PositionCreate(BaseModel):
    title: str
    department: str = ""
    description: str = ""
    required_skills: list[str] = []
    experience_min_years: float = 0
    salary_range_min: float | None = Field(default=None, ge=0)
    salary_range_max: float | None = Field(default=None, ge=0)

    @model_validator(mode="after")
    def _check_salary_range(self) -> "PositionCreate":
        if (
            self.salary_range_min is not None
            and self.salary_range_max is not None
            and self.salary_range_min > self.salary_range_max
        ):
            raise ValueError("salary_range_min no puede ser mayor que salary_range_max")
        return self


class PositionResponse(BaseModel):
    id: UUID
    title: str
    department: str | None
    description: str | None
    required_skills: list[str] | None
    experience_min_years: float | None
    salary_range_min: float | None
    salary_range_max: float | None
    status: str
    candidate_count: int = 0

    model_config = ConfigDict(from_attributes=True)


class CandidateResponse(BaseModel):
    # AI.SCO — `score` y `score_breakdown` no se exponen por cumplimiento
    # Anexo III AI Act. Ver `docs/ai_act_scoping.md`.
    id: UUID
    position_id: UUID | None
    name: str
    email: str | None
    phone: str | None
    skills: list[str] | None
    experience_years: float | None
    languages: list | None
    education: str | None
    summary: str | None
    status: str
    cv_file_path: str | None

    model_config = ConfigDict(from_attributes=True)


class StatusUpdate(BaseModel):
    status: Literal["new", "reviewed", "shortlisted", "rejected", "hired"]
